### ISSUE TITLE
GetDirectorySynchronizationCookie() returns empty cookie

### ISSUE THREAD

**Issue by EvanMulawski** [dotnet/runtime#49376]
### Description

`System.DirectoryServices.DirectorySynchronization.GetDirectorySynchronizationCookie()` returns an invalid, 0-byte cookie on `net5.0`. A valid cookie is returned when running the same code on `netcoreapp3.1`.

Repro: https://github.com/EvanMulawski/DotNetDirectoryServicesDirSyncIssue

### Configuration

- `net5.0` (5.0.3)
- Windows 10 64-bit 1909

### Regression?

Works in:

- `netcoreapp3.1` (3.1.12)
- .NET Framework 4.8

**Comment by Thomas-Avery** [comment]
I am experiencing the same 0-byte cookie problem; however, it seems isolated to the version of System.DirectoryServices not net5.0 vs netcoreapp3.1.


| Target Framework       |Runtime     |System.DirectoryServices NuGet Package Version     |Result     |
| :------------- | :----------: | -----------: | :-----------: |
|  netcoreapp3.1 | 3.1.14   | 5.0.0    | FAIL    |
|  netcoreapp3.1 | 3.1.14   | 4.7.0    | PASS    |
|  net5.0 | 5.0.5   | 5.0.0    | FAIL    |
|  net5.0   | 5.0.5 | 4.7.0    | PASS

**Comment by Thomas-Avery** [comment]
After some more research it looks like in the past the DirectorySearcher class, in the FindAll() private method, it would set an internal SearchResultCollection searchResult property. 

https://github.com/dotnet/runtime/blob/60d1224ddd68d8ac0320f439bb60ac1f0e9cdb27/src/libraries/System.DirectoryServices/src/System/DirectoryServices/DirectorySearcher.cs#L677-L679

This internal searchResult property is what is used to set the DirsyncCookie when you call directorySearcher.DirectorySynchronization.

https://github.com/dotnet/runtime/blob/eb7ef11d84c309f4f041ecbc1642d25d8d11a3ec/src/libraries/System.DirectoryServices/src/System/DirectoryServices/DirectorySearcher.cs#L503-L508


Now in the FindAll() code it just returns a new SearchResultCollection without populating the internal searchResult property. Which I believe is the root of the problem.
https://github.com/dotnet/runtime/blob/eb7ef11d84c309f4f041ecbc1642d25d8d11a3ec/src/libraries/System.DirectoryServices/src/System/DirectoryServices/DirectorySearcher.cs#L664

**Comment by joperezr** [comment]
@Thomas-Avery thanks for the great investigation! Have you had a chance to revert that change to again set searchResult property and validate that this fixes the issue you are facing?

**Comment by joperezr** [comment]
cc: @tquerec @grubioe Do you mind taking a look at this one since it is in S.DS?

**Comment by joperezr** [comment]
I don't think that we'll get to fix this in 6.0.0 so moving it to 7.0.0 for now.

**Comment by arora-kushal** [comment]
The same error is happening for:

```
Target Framework : net 5.0/netcoreapp3.1
Runtime : 5.0.13 (SDK - 5.0.404)
System.DirectoryServices NuGet Package Version: 4.7.0
Result: FAIL
```

**Workaround**: I have used ```FindOne()``` method, instead of ```FindAll()``` to get cookie.

**Comment by knoxi** [comment]
+1
run into the same on .net 6, had to rollback `System.DirectoryServices` to 4.7.0 to get it working again

**Comment by joperezr** [comment]
@knoxi did the workaround provided above by @arora-kushal not work for you?

**Comment by knoxi** [comment]
@joperezr I didn't tried this one, since I'm not sure if the behavior is still the same when using `FindOne()` instead of `FindAll()` because I'm using a query with `FindAll()`. Maybe I should give it a try.

**Comment by hrrrrustic** [comment]
This was fixed by https://github.com/dotnet/runtime/pull/79093, could be closed

**Comment by buyaa-n** [comment]
Yep, it is fixed and the fix ported to 6.0/7.0

**Comment by hrrrrustic** [comment]
@buyaa-n I found one more similar issue #59756 that can be closed
